azure-storage-cpp-lite/include/retry.h

```diff
@@ -67,8 +68,11 @@ namespace microsoft_azure {
             retry_info evaluate(const retry_context &context) const override {
                 if (context.numbers() == 0) {
                     return retry_info(true, std::chrono::seconds(0));
-                } else if (context.numbers() < 100 && can_retry(context.result())) {
-                    return retry_info(true, std::chrono::seconds(5+rand()%5));
+                } else if (context.numbers() < 60 && can_retry(context.result())) {
```

Comment:
I propose we reduce this to 10-20 retries, because with backoff implemented, retries will now take much more longer... Nothing should retry for more than a few minutes... What do you all think ?

Reply:
When Adam and I were discussing, we were thinking we reaaaally don't want to fail if at all possible since it can be tough for the user to figure out what to do. These retries will max out at a little less than an hour, which we thought would be more important for large data transfers

Reply:
As it stands now, yes.

Reply:
Depends on whether failure is worse than delay... the issue for me is that consumers have very little they can do in the form of error-recovery (especially for lift-and-shift apps, written assuming a local HD).  If there's any chance at all of recovery, it might be worth it.  OTOH, if there's a nonrecoverable network issue, failing faster is always better...

How about we change the base to 1.2 (see below), and set the max retry count to 26?  If my math is right, this would give a max of 10 minutes before retries are done, give or take from jitter

Reply:
I think it is a good idea. If it takes more than 10 minutes, user should consider a different solution (sharding data across storage accounts)

Reply:
So, I just realized - in this case, 10 minutes is all the time spent waiting.  This doesn't include time spent inside the actual request.  Worst case - each try in a non-data-transfer operation takes the full 30 seconds to fail before retry, that adds 30 seconds to each, which would total 10 min + 26/2 = 23 minutes total.  (Data-transfer operations are practically unbounded, pathological worst case at 55 days per try, but I'm not as concerned about this.)

Reply:
unfortunately, the user has no way to terminate the operation if this happened. he cannot even kill his process. may be it's better to make retries configurable and fail fast by default?

Reply:
Making retries configurable is a good idea, but I think we should do that in a separate commit/PR, if that's ok?